Sources/TensorFlow/Core/Runtime.swift

```diff
@@ -404,13 +401,15 @@ internal extension _ExecutionContext {
     /// Synchronously execute the body, preventing asynchronous computation from corrupting the
     /// context data.
     private func sync<Result>(execute body: () throws -> Result) rethrows -> Result {
-        let lockStatus = pthread_mutex_lock(&mutex)
+        let lockStatus = mutex.acquire()
         internalConsistencyCheck(lockStatus == 0)
         defer {
-            let unlockStatus = pthread_mutex_unlock(&mutex)
+            let unlockStatus = mutex.release()
             internalConsistencyCheck(unlockStatus == 0)
+#if !os(Windows)
             // Create a cancellation point.
             pthread_testcancel()
```

Comment:
Ignorant question: does it not make sense to move `pthread_testcancel()` to `Mutex.release`? It seems unfortunate that callers of `Mutex.release` still need an extra `#if !os(Windows)` directive with `pthread_testcancel()`.

Reply:
Im entirely happy to do that.  The reasoning behind not doing so immediately is that should every release be associated with a cancellation point?

Reply:
> The reasoning behind not doing so immediately is that should every release be associated with a cancellation point?

I'm clueless. Maybe @pschuh has an opinion?

Reply:
Okay, did a bit of code archeology with @pschuh ... I think that we came to the conclusion that we do not have a strong reason to keep this functionality at this time.  I can do a follow up change to remove the cancellation point support.